### PR TITLE
chore(deps): update required pnpm version

### PR DESCRIPTION
**Pull request by jmetev1**
Checklist:

- [ x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [ x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [ x] My pull request targets the `main` branch of freeCodeCamp.
- [x ] I have tested these changes either locally on my machine, or GitPod.

Closes na 

I think renovate bot and probably github actions are automatically updating to latest pnpm. You can see the bot bumping the lockfile version [here](https://github.com/freeCodeCamp/freeCodeCamp/pull/50597/files#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1) So when I ran pnpm install locally, it failed due to this until i updated pnpm to 8.6.0. This change to package.json will give people a warning when they run `pnpm install` if they their pnpm version is out of date.

### CODE REVIEW

**Comment by Sboonny** [comment]
Is this PR is trying to solve, the issue in the image below, it will block development through Gitpod, until Gitpod version is 8.6.x.

image

**Comment by jmetev1** [comment]
Or in general it seems like we would like gitpod to be on same pnpm version as renovate bot and our ci right? So either we have to raise gitpod pnpm version or lower version in ci right? I think I see how to change gitpod version and ci version. But renovate bot seems more mysterious to me and i don't think have access.

**Comment by Sboonny** [comment]
You are misunderstanding, Gitpod has the same issue 😞, but the fix will introduce another issue for it too.

I am assigning Shaun to handle this, when he got time. Thank you for bringing this to our attention.

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "private": true,
   "engines": {
     "node": ">=16",
-    "pnpm": "8"
+    "pnpm": "8.6.x"
```
Comment (raisedadead) [inline package.json:9]: Undo this change

File: docs/how-to-setup-freecodecamp-locally.md [hunk 2]
```diff
@@ -69,7 +69,7 @@ We primarily support development on Linux and Unix-based systems like Ubuntu and
 | Prerequisite                                                                                  | Version | Notes                                                                                       |
 | --------------------------------------------------------------------------------------------- | ------- | ------------------------------------------------------------------------------------------- |
 | [Node.js](http://nodejs.org)                                                                  | `18.x`  | We use the "Active LTS" version, See [LTS Schedule](https://nodejs.org/en/about/releases/). |
-| [pnpm](https://pnpm.io/installation)                                                          | `8.x`   | -                                                                                            |
+| [pnpm](https://pnpm.io/installation)                                                          | `8.6.x`   | -                                                                                            |
```
Comment (raisedadead) [inline docs/how-to-setup-freecodecamp-locally.md:72]: Undo this change. 

Instead, add the error text and troubleshooting info here or in the FAQ later.

The notes bit in the table should read like so:

```markdown
If you see the error: 'Your ... <add the full text>' See [troubleshooting](link to the troubleshooting).
```

The troubleshooting FAQ should read like so:

```markdown
pnpm updates the lock file. If do not have the latest of version of `pnpm` then simply update your `pnpm` by either following the installation [instructions here](https://pnpm.io/installation) OR run `npm install -g pnpm@latest`
```

**Review by raisedadead — CHANGES_REQUESTED** [review]
No need to be this specific with the semver. The issue is you (as a developer) are on an outdated version of pnpm and need to bump it up to the latest.

The quickest way is to `npm install -g pnpm@latest` OR read other specific instructions in their docs.

It doesn't matter what IDE you use. Because the CI/CD is on a higher version, as a dev, you need to bump it up in your environment to match what we use in our deployments.

File: docs/how-to-setup-freecodecamp-locally.md (hunk 2 above)
Comment (raisedadead) [reply docs/how-to-setup-freecodecamp-locally.md:72]: For GitPod users separate changes can be done by adding a pre/post setup steps in the config to achieve same results.
